##### Categorical / Identifier Variables

In [1]:
%run '../lib/libraries.ipynb'

dython              0.6.1


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


tinydb              4.2.0


You should consider upgrading via the '/Users/duane.hinkley/PycharmProjects/c772-capstone-project/venv/bin/python -m pip install --upgrade pip' command.


In [2]:
dfFlt  = load_df('dfFlt')
dfDesc = load_df('dfDesc')
types = get_var_types()

In [3]:

unique_nulls(dfFlt, types['identifierVars'])

assessment_id
+------+----+
|unique|null|
+------+----+
|   329|   0|
+------+----+

assessment_instance_attempt_id
+------+----+
|unique|null|
+------+----+
|  8483|3264|
+------+----+

assessment_instance_id
+------+----+
|unique|null|
+------+----+
|   615|   0|
+------+----+

assessment_item_response_id
+------+-----+
|unique| null|
+------+-----+
| 64368|15710|
+------+-----+

learner_assigned_item_attempt_id
+------+----+
|unique|null|
+------+----+
| 79689|   0|
+------+----+

learner_assignment_attempt_id
+------+----+
|unique|null|
+------+----+
|  8855|   0|
+------+----+

learner_id
+------+----+
|unique|null|
+------+----+
|  1126|   0|
+------+----+

org_id
+------+----+
|unique|null|
+------+----+
|     3|   0|
+------+----+

section_id
+------+----+
|unique|null|
+------+----+
|    46|   0|
+------+----+



Totals
- org_id: 3
  - ID per organization
- section_id: 46
  - ID per section/school
- learner_id: 1126
  - ID per learner/student
- assessment_id: 329
  - ID per each assessment/quiz.
- assessment_instance_id: 615
  - ID of each time a section uses the assessment
- assessment_instance_attempt_id: 8483 null: 3264
  - ID of each time an assessment is attempted by a student
- assessment_item_response_id: 64368 null: 15710
  - ID of each response/answer to an assessment question
- learner_assignment_attempt_id: 8855
  - ID of each learner attempt of an assigned assessment
- learner_assigned_item_attempt_id: 79689
  - ID of each learner attempt of a item/question on an assigned assessment

In [4]:
add_todo('Investigate 15710 null assessment_item_response_id')
